client/my-sites/stats/hooks/use-intervals.tsx

```diff
@@ -58,15 +65,18 @@ const getGatedIntervals = createSelector(
 			};
 		}, {} );
 	},
-	Object.values( intervals ).map(
-		( { statType } ) =>
-			( state: object, siteId ) =>
-				shouldGateStats( state, siteId, statType )
-	)
+	( state, siteId, intervals ) => {
```

Comment:
Does this actually work? If it returns a new object on every call, then it's likely the cache always gets busted/invalidated. Let's check the implementation of `createSelector` to confirm pls.

Reply:
`createSelector` does [a shallow equality check](https://github.com/Automattic/wp-calypso/blob/573de31f0d27a34153e746216905cf2cdccefad0/packages/state-utils/src/create-selector/index.ts#L103) using the `@wordpress/is-shallow-equal` and as far as I can tell it [compares the values of the array rather than then array reference](https://github.com/WordPress/gutenberg/blob/7f6a627a028f6c5013dc97df316e04e8fc39a4ef/packages/is-shallow-equal/src/index.js#L23).

However, I haven't looked too much in details so it might be worth double checking.

Reply:
Thanks for the help, @chriskmnds and @yuliyan! The memoization issue is resolved now.